api/handlers/devicehandler.py

```diff
@@ -0,0 +1,124 @@
+import datetime as dt
+
+from .. import base
+from .. import config
+from .. import util
+from ..dao import containerstorage, APIPermissionException, APINotFoundException
+from ..types import Origin
+
+log = config.log
+
+Status = util.Enum('Origin', {
+    'ok':       'ok',       # Device's last seen time is shorter than expected interval for checkin, no errors listed.
+    'missing':  'missing',  # Device's last seen time is longer than the expected interval for checkin, but no errors listed.
+    'error':    'error' ,   # Device has errors listed.
+    'unknown':  'unknown'   # Device did not set an expected checkin interval.
+})
+
+
+def require_login(handler_method):
+    """
+    A decorator to ensure the request is not a public request.
+
+    Accepts superuser and non-superuser requests.
+    Accepts drone and user requests.
+    """
+    def check_login(self, *args, **kwargs):
+        if self.public_request:
+            raise APIPermissionException('Login required.')
+        return handler_method(self, *args, **kwargs)
+    return check_login
+
+def require_superuser(handler_method):
+    """
+    A decorator to ensure the request is made as superuser.
+
+    Accepts drone and user requests.
+    """
+    def check_superuser(self, *args, **kwargs):
+        if not self.superuser_request:
+            raise APIPermissionException('Superuser required.')
+        return handler_method(self, *args, **kwargs)
+    return check_superuser
+
+def require_drone(handler_method):
+    """
+    A decorator to ensure the request is made as a drone.
+
+    Will also ensure superuser, which is implied with a drone request.
+    """
+    def check_drone(self, *args, **kwargs):
+        if self.origin.get('type', '') != Origin.device:
+            raise APIPermissionException('Drone request required.')
+        if not self.superuser_request:
+            raise APIPermissionException('Superuser required.')
+        return handler_method(self, *args, **kwargs)
+    return check_drone
```

Comment:
Why not take these three closure generators and move them to util?

This looks like a decent pattern you're starting, might as well use it everywhere (eventually).